### PR TITLE
[Fix] enable visualization of the demo results online

### PR DESCRIPTION
**Pull request by wHao-Wu**
* Enabling visualization of the pcd demo results online
* Enabling visualization of the multi-modality demo results online

**Risk**: Whether to simultaneously visualize the point and image results online when visualizing multi-modality demo?

### CODE REVIEW

**Comment by wHao-Wu** [comment]
## PCD demo
### SECOND:
kitti_000008_online

### VoteNet:
sunrgbd_000017_online

## Multi-modality demo
### MVX:
kitti_000008_online

### ImVoteNet:
sunrgbd_000017_online

**Comment by Wuziyi616** [comment]
@wHao-Wu Are we able to visualize img and points at the same time? As far as I know, we need to press `q` to exit open3d.vis and mmcv.imshow(). How can you show them simultaneously? I know there is a so-called non-blocking mode in opencv but I haven't tried that before.

BTW, in multi-modality browse_dataset, I first show 3D point results, and then show 2D img results when the user exits 3D window.

File: mmdet3d/core/visualizer/show_result.py [hunk 1]
```diff
@@ -87,10 +90,8 @@ def show_result(points, gt_bboxes, pred_bboxes, out_dir, filename, show=True):
             vis.add_bboxes(bbox3d=pred_bboxes)
         if gt_bboxes is not None:
             vis.add_bboxes(bbox3d=gt_bboxes, bbox_color=(0, 0, 1))
-        vis.show()
-
-    result_path = osp.join(out_dir, filename)
-    mmcv.mkdir_or_exist(result_path)
+        show_path = osp.join(result_path, f'{filename}_online.png')
+        vis.show(show_path)
```
Comment (Wuziyi616) [inline mmdet3d/core/visualizer/show_result.py:94]: I don't think we should save the online results every time, at least let's pass a parameter to determine whether to save? Like what I've done [here](https://github.com/Wuziyi616/mmdetection3d/blob/multi_dataset_browse/mmdet3d/core/visualizer/show_result.py#L102-105)?

Comment (wHao-Wu) [reply mmdet3d/core/visualizer/show_result.py:94]: Thanks for your advice. I have also considered the problem of mandatory saving online results. I will modify the corresponding operation lately.

**Comment by wHao-Wu** [comment]
> @wHao-Wu Are we able to visualize img and points at the same time? As far as I know, we need to press `q` to exit open3d.vis and mmcv.imshow(). How can you show them simultaneously? I know there is a so-called non-blocking mode in opencv but I haven't tried that before.
> 
> BTW, in multi-modality browse_dataset, I first show 3D point results, and then show 2D img results when the user exits 3D window.

**Comment by wHao-Wu** [comment]
> @wHao-Wu Are we able to visualize img and points at the same time? As far as I know, we need to press `q` to exit open3d.vis and mmcv.imshow(). How can you show them simultaneously? I know there is a so-called non-blocking mode in opencv but I haven't tried that before.
> 
> BTW, in multi-modality browse_dataset, I first show 3D point results, and then show 2D img results when the user exits 3D window.

I am sorry I have not explained the problem clearly. I mean that maybe we don't need to visualize the points results every time because we have shown the final results in which the detected boxes have been drawn in images.

**Comment by Wuziyi616** [comment]
> > @wHao-Wu Are we able to visualize img and points at the same time? As far as I know, we need to press `q` to exit open3d.vis and mmcv.imshow(). How can you show them simultaneously? I know there is a so-called non-blocking mode in opencv but I haven't tried that before.
> > BTW, in multi-modality browse_dataset, I first show 3D point results, and then show 2D img results when the user exits 3D window.
> 
> I am sorry I have not explained the problem clearly. I mean that maybe we don't need to visualize the points results every time because we have shown the final results in which the detected boxes have been drawn in images.

Maybe we can ask what @ZwwWayne thinks. In my opinion, both 3d points and 2d image are input data so it would be better if we show both?

**Comment by baheytharwat** [comment]
> @wHao-Wu Are we able to visualize img and points at the same time? As far as I know, we need to press `q` to exit open3d.vis and mmcv.imshow(). How can you show them simultaneously? I know there is a so-called non-blocking mode in opencv but I haven't tried that before.
> 
> BTW, in multi-modality browse_dataset, I first show 3D point results, and then show 2D img results when the user exits 3D window.

@Wuziyi616 , Is there a way to update the open3d visualization window without pressing `q` each time? I am using `show_result_meshlab` function from the demo example
I just want to display the detection results of multiple point clouds sequentially without pressing any key

**Pull request merged**